### ISSUE TITLE
BERT Model for Tokenization and Initialization of embedding layer

### ISSUE THREAD

**Issue by kr-sundaram** [bentrevett/pytorch-seq2seq#114]
Thank you very much for all your tutorials! :)

In your tutorial6, You have used spacy tokenizer for tokenization and initialized the embedding layer with xavier_uniform as stated below.

def initialize_weights(m):
    if hasattr(m, 'weight') and m.weight.dim() > 1:
        nn.init.xavier_uniform_(m.weight.data)

I am not sure how to go about using BERT model from huggingface transformer for both tokenization and word-embeddings in order to make vocab open for rare or unseen words and initialize the embedding layer with pre-trained BERT embedding for both encoder and decoder.

Could you please help me what needs to be done in order to achieve these two goals?

Any tip or tutorial will be helpful!

Thanks for your patience! :)

**Comment by yugaljain1999** [comment]
For tokenization just use BertTokenizer-
`from transformers.tokenize import BertTokenizer`
and word-embeddings is already initialized in BERT model which you are trying to import for downstream tasks.

If you have still some doubts, feel free to ask.
